src/input/mod.rs

```diff
@@ -327,15 +327,51 @@ impl State {
                 )
             },
         ) else {
+            self.remove_all_bind_repeat_timers();
             return;
         };
 
-        // Filter actions when the key is released or the session is locked.
-        if !pressed {
-            return;
+        if pressed {
+            self.handle_bind(bind.clone());
+
+            match self.niri.bind_repeat_timers.entry(bind.key) {
+                Entry::Vacant(entry) => {
+                    let repeat_delay_timer = Timer::from_duration(Duration::from_millis(
+                        self.niri.config.borrow().input.keyboard.repeat_delay as u64,
+                    ));
+
+                    let token = self
+                        .niri
+                        .event_loop
+                        .insert_source(repeat_delay_timer, move |_, _, state| {
+                            state.handle_bind(bind.clone());
+
+                            TimeoutAction::ToDuration(Duration::from_millis(max(
+                                1000 / state.niri.config.borrow().input.keyboard.repeat_rate as u64,
```

Comment:
Compute this TimeoutAction once, before inserting the timer, then just use it. Also instead of a max() on the 1000 / rate, instead check `if rate == 0 { TimeoutAction::Drop } else { TimeoutAction::ToDuration(...) }`.

Reply:
So, I don't think there should be any hardcoded limit here. E.g. in GNOME spawn repeats just like anything else, and it's useful if you want to stress-test the compositor with a bunch of terminals. Instead, what I was thinking, is to add a property to binds, something like `repeat=false` (same place as the existing `cooldown-ms=150`) that will exclude it from key repeat entirely.